### PR TITLE
Added KeySignatureError and tests. Also added bidict dependency.

### PR DESCRIPTION
**Pull request by cjappl**

### CODE REVIEW

File: mido/midifiles/meta.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
      - expose _key_signature_encode/decode?
 """
 from __future__ import print_function, division
+from bidict import bidict
```
Comment (cjappl) [inline mido/midifiles/meta.py]: I can definitely remove this if need be. I felt at least a little weird adding a dependancy for it.

This is a great package, especially for this "encode/decode" thing you have going on here with the dictionaries.

The bidict basically can have keys as values, and values as keys, so you can access the "inverse" of the dictionary with "my_bidict.inv", similar to what was being done in "reverse_table".

Again, happy to revert out this bidict change, just wanted to show what it was capable of!

**Comment by cjappl** [comment]
From #105

**Comment by cjappl** [comment]
@olemb Sorry to bug you, but I, unfortunately, wrote this on my master branch. I wanted to clear this up before moving on to the tox/dev dependencies task.

Cheers!

**Comment by olemb** [comment]
Sure thing.

I ran into a problem though. `./setup.py` won't run unless `bidict` is installed:

```
Traceback (most recent call last):
  File "./setup.py", line 4, in <module>
    import mido
  File "/home/olemb/src/mido/mido/__init__.py", line 100, in <module>
    from .midifiles import (MidiFile, MidiTrack, merge_tracks,
  File "/home/olemb/src/mido/mido/midifiles/__init__.py", line 1, in <module>
    from .meta import MetaMessage, UnknownMetaMessage, KeySignatureError
  File "/home/olemb/src/mido/mido/midifiles/meta.py", line 14, in <module>
    from bidict import bidict
ImportError: No module named bidict
```
This is because it imports things like `__version__` and `__author__` from the top level module, which also imports `meta.py`.

This means pip will fail as well so you can't actually install Mido unless bidict is already installed. There must be a fix for this that doesn't involve updating version etc. in two places.

**Comment by cjappl** [comment]
Got a solution, will submit it with the other PR

**Comment by olemb** [comment]
Cool!

**Comment by cjappl** [comment]
see #123

**Pull request merged**